server/src/printers/NativeMakefilePrinter.cpp

```diff
@@ -469,13 +469,15 @@ namespace printer {
                     }
                     linkCommand.addFlagToBegin("$(LDFLAGS)");
                     if (isExecutable) {
-                        linkCommand.addFlagToBegin(RELOCATE_FLAG);
+                        linkCommand.addFlagToBegin(transformExeToLib ? SHARED_FLAG : RELOCATE_FLAG);
                     }
                 }
-                return isExecutable ? stringFormat("%s && objcopy --redefine-sym main=old_main %s",
-                                                   linkCommand.toStringWithChangingDirectory(),
-                                                   linkCommand.getOutput().string()) :
-                                      linkCommand.toStringWithChangingDirectory();
+                if (isExecutable && transformExeToLib) {
+                    return stringFormat("%s && objcopy --redefine-sym main=old_main %s",
```

Comment:
I thought we already redefined `main` function in wrappers using `#define main main__`, didn't we?